api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseEncounterDao.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.dao.impl;
+
+import static org.hl7.fhir.r4.model.Encounter.SP_DATE;
+
+import javax.annotation.Nonnull;
+
+import org.openmrs.Auditable;
+import org.openmrs.OpenmrsObject;
+
+/**
+ * Base class for Encounter-related DAO objects. This helps standardise the logic used to sort
+ * objects that represent an Encounter {@link org.openmrs.Encounter}.
+ *
+ * @param <T> The OpenMRS object that represents an Encounter.
+ */
+public abstract class BaseEncounterDao<T extends OpenmrsObject & Auditable> extends BaseFhirDao<T> {
+	
+	@Override
+	protected String paramToProp(@Nonnull String param) {
+		switch (param) {
+			case SP_DATE:
+				return "encounterDatetime";
+			default:
+				return null;
+		}
+	}
```

Comment:
Why not override your method here; https://github.com/openmrs/openmrs-module-fhir2/blob/master/api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirEncounterDaoImpl.java

Reply:
This seems better. I'll do the changes.